### PR TITLE
Convert to 2.2 class mixins - patches, queries, and storage

### PR DESCRIPTION
**Pull request by maier49**
**Type:** feature

**Description:** 
Converts part of the stores repo to use TypeScript classes instead of `dojo/compose`. Doesn't replace any mixins so 2.2 class mixins are not used.

Relates to: #108
Please review this checklist before submitting your PR:

* [x] There is a related issue
* [x] All contributors have signed a CLA
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] The code passes the CI tests
* [x] Unit or Functional tests are included in the PR
* [x] The PR increases or maintains the overall unit test coverage percentage
* [x] The code is ready to be merged

### CODE REVIEW

File: tests/intern.ts
Comment (dylans) [inline tests/intern.ts:59]: Do we still need rxjs here? Probably not related to this PR, but might as well clean it up since this PR is cleaning up many things...

File: tests/unit/support/AsyncStorage.ts
Comment (dylans) [inline tests/unit/support/AsyncStorage.ts:22]: interesting that this now needs compose?

**Review by dylans — APPROVED** [review]
Not the most thorough review, but it looks solid. There are a few things I don't quite understand yet (e.g. why we still have things like createSort and createFilter), but I assume there's a reason that I'm not following yet, so I think it has my approval. You may want to wait for someone else to approve as well.

File: tests/unit/support/AsyncStorage.ts
Comment (maier49) [reply tests/unit/support/AsyncStorage.ts:22]: This is a temporary hack as I haven't updated this test implementation(on this branch). It's just wrapping the class in a call to compose to make it a factory so it can be extended.

File: tests/intern.ts
Comment (maier49) [reply tests/intern.ts:59]: No we don't need it. Good catch I'll clean that up.

**Comment by maier49** [comment]
@dylans To your general comment about `createSort`, `createFilter`, etc., that's probably part of a larger discussion about what moving back to TypeScript classes will mean for some of our patterns. Those are just factory functions and aren't actually leveraging `compose`. I can implement them as classes if we now want to avoid factories all together and strongly prefer classes. I don't have a strong feeling about it either way.

@agubler Any thoughts on the above?

**Comment by agubler** [comment]
I think you're right regarding the need for a broader discussion about the general patterns used across Dojo 2 post moving to Classes from compose. As for simple factory functions, I actually don't mind them as a pattern - sometimes using classes is probably unnecessary (this is kind of proved by the fact that they were not compose factories anyway).

We have functions in widget-core like `w` and `v` the operate in a similar fashion.

**Pull request merged**
